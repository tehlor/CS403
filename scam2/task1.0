(for-loop (range 0 9 5) (lambda (x) (inspect x)))
