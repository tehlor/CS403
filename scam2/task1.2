(for-loop (range 74 139 16) (lambda (z) (inspect (+ -9 z))))
