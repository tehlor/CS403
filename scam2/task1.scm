(define (main)
    (setPort (open (getElement ScamArgs 1) 'read))
    (readArgs)
)

(define (readArgs)
    (define env this)
    (define (iter expr)
        (if (not (eof?))
            (begin (eval expr env) 
                (iter (readExpr))
            )
        )
    )
    (iter (readExpr))
)

(define (range a b i)
    (if (< a b)
        (cons a (range (+ a i) b i))
        nil
    )
)

(define (for-loop result function)
    (define (iter i len)
        (cond
            ((< i len)
                (function (getElement result i))
                (iter (+ i 1) len)
            )
        )
    )
    (iter 0 (length result))
)