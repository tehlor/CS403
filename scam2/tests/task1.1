(println (range 1 11 3))
