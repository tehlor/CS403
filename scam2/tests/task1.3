(for-loop (range 32 110 17) (lambda (z) (inspect (* 5 z))))
