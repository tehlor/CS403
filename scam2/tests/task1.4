(for-loop (range 41 132 17) (lambda (z) (inspect (+ 9 z))))
