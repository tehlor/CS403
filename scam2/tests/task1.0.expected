x is 0
x is 5
